src/launcher.cxx

```diff
@@ -60,10 +67,15 @@ bool raft_launcher::shutdown(size_t time_limit_sec) {
     raft_instance_->shutdown();
     raft_instance_.reset();
 
-    if (asio_listener_) {
-        asio_listener_->stop();
-        asio_listener_->shutdown();
+
+    for (const auto & asio_listener : asio_listeners_)
```

Comment:
It'd be more consistent to close listeners before `raft_instance_->shutdown()` i.e. uninitialize it in reverse order. Otherwise, it can happen that we'll accept a connection but the object which should serve it will be already destroyed

Reply:
Probably we can contribute to NuRaft with this fix

Reply:
`shared_ptr` is used + requests are skipped if the server is shutdown.
I agree it look unnatural but some messages are being finalized and canceled during shutdown so I think it's better to keep the asio service + listeners alive as much as possible.